jenkins_job_wrecker/modules/publishers.py

```diff
@@ -75,33 +75,166 @@ def fingerprinter(top, parent):
         elif element.tag == 'presendScript':
             ext_email['presend-script'] = element.text
+        elif element.tag == 'postsendScript':
+            ext_email['postsend-script'] = element.text
         elif element.tag == 'sendTo':
             ext_email['send-to'] = element.text
         elif element.tag == 'configuredTriggers':
-            print("IGNORED configuredTriggers in email-ext")
+            trigger_prefix = "hudson.plugins.emailext.plugins.trigger."
+            trigger_dict = {
+                "AlwaysTrigger": "always",
+                "UnstableTrigger": "unstable",
+                "FirstFailureTrigger": "first-failure",
+                "FirstUnstableTrigger": "first-unstable",
+                "NotBuiltTrigger": "not-built",
+                "AbortedTrigger": "aborted",
+                "RegressionTrigger": "regression",
+                "FailureTrigger": "failure",
+                "SecondFailureTrigger": "second-failure",
+                "ImprovementTrigger": "improvement",
+                "StillFailingTrigger": "still-failing",
+                "SuccessTrigger": "success",
+                "FixedTrigger": "fixed",
+                "FixedUnhealthyTrigger": "fixed-unhealthy",
+                "StillUnstableTrigger": "still-unstable",
+                "PreBuildTrigger": "pre-build"
+            }
+            provider_dict = {
+                "hudson.plugins.emailext.plugins.recipients.UpstreamComitterRecipientProvider": "upstream-committers",
+                "hudson.plugins.emailext.plugins.recipients.DevelopersRecipientProvider": "developers",
+                "hudson.plugins.emailext.plugins.recipients.RequesterRecipientProvider": "requester",
+                "hudson.plugins.emailext.plugins.recipients.CulpritsRecipientProvider": "culprits",
+                "hudson.plugins.emailext.plugins.recipients.ListRecipientProvider": "recipients",
+                "hudson.plugins.emailext.plugins.recipients.FailingTestSuspectsRecipientProvider":
+                    "failing-test-suspects-recipients",
+                "hudson.plugins.emailext.plugins.recipients.FirstFailingBuildSuspectsRecipientProvider":
+                    "first-failing-build-suspects-recipients"
+            }
+            trigger_result = {}  # To be able to print all on the output yaml
+            for key in trigger_dict.keys():
+                trigger_result[key] = False
+            # JJB implementation only uses the default content-type/subject/message
+            # and uses those in the trigger elements. Does not take information
+            # from trigger element's content. However, sendTo must be taken from
+            # one of the triggers, in this case, first trigger will be used in order
+            # to retrieve sendTo information. If there are multiple triggers,
+            # NotImplementedError will be raised to make sure JJW does not cause
+            # information loss.
```

Comment:
Please revise this comment - my understanding (of the implementation) is that JJW can do the conversion only if triggers use the defaults for subject and body + recipients providers match the ones configured globally. Otherwise NotImplementedError is raised to avoid loosing information. The comment does not make it clear what conditions need to occur for the conversion to happen properly. And this is also the key for anyone refactoring or changing this code in the future.

What do you think?

Reply:
How is it now?

Reply:
Much better, thanks.